Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -121,6 +150,27 @@ Ref<SharedBuffer> FragmentedSharedBuffer::makeContiguous() const
     return SharedBuffer::create(WTFMove(combinedData));
 }
 
+#if PLATFORM(COCOA)
+auto FragmentedSharedBuffer::toIPCData() const -> FragmentedSharedBuffer::FragmentedSharedBufferData
+{
+    size_t bufferSize = size();
+    if (!bufferSize)
+        return std::span<const uint8_t> { };
+
+    if (bufferSize < minimumPageSize) {
+        if (m_segments.isEmpty())
+            return std::span<const uint8_t> { };
+        if (m_segments.size() == 1)
+            return std::span { m_segments[0].segment->data(), m_segments[0].segment->size() };
+        auto combinedData = combineSegmentsData(m_segments, m_size);
+        return combinedData.span();
```

Comment:
Sorry, I think I gave you incorrect code here. Now that I think of it, it is probably not safe to return a span<> to a local Vector. The Vector goes out of scope after returning and the span points to bad data. We have to write this differently somehow. I'll think about it.

Reply:
I think we could return a `Vector<std::span<const uint8_t>>` instead. Then on sender side, you iterate over each segment and append to that vector. This is more efficient since we don't need to copy the bytes. Also, we no longer need a local Vector so no lifetime issue.

On recipient side, you call `append()` already anyway so you can append() the spans one by one. WDYT?

Reply:
Do we want `Vector<std::span<const uint8_t>>` or just `Vector<uint8_t>`?